docs/whatsnew/2.0.rst

```diff
@@ -46,6 +46,11 @@ change has been made to behavior for Python 2.
 Other significant changes
 =========================
 
+Convolution has undergone a significant overhaul.  They keyword arguments have
+changed and the behavior of `~astropy.convolution.convolve` are no longer the
+same as in versions 1.3 and earlier (however,
+`~astropy.convolution.convolve_fft`'s behavior remains unchanged.  The details
+are given on the :ref:`convolution <astropy_convolve>` documentation page.
```

Comment:
Maybe also mention that un-normalized kernels are now possible?  That's a distinct new feature, really.